src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -286,7 +290,27 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
                 }
             }
 
-            return validTagName && validAttributeNames;
+            var validParentTagName = ValidateParentTagName(attribute.ParentTag, errorSink);
+
+            return validTagName && validAttributeNames && validParentTagName;
```

Comment:
Is this intentionally not short-circuited so invalid parent tag errors get added to error sink?